lyngk_board.js

```diff
@@ -0,0 +1,210 @@
+    var ctx=board.getContext("2d");
+    ctx.beginPath();
+    ctx.arc(40, piecePosition * 62, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[letterToColor[color]];
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  function drawClaimedPiece(color, boardNumber, piecePosition) {
+    boardLabel = '';
+    switch(boardNumber) {
+      case 1:
+        boardLabel = "player1canvas";
+        break;
+      case 2:
+        boardLabel = "player2canvas";
+        break;
+    }
+    var board =document.getElementById(boardLabel);
+    var ctx=board.getContext("2d");
+    ctx.beginPath();
+    ctx.arc(piecePosition * 29, 25, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[letterToColor[color]];
+    ctx.fill();
+    ctx.stroke();
+  }
+
+  function populateClaimedCanvases() {
+    var unclaimedBoard = document.getElementById("unclaimedCanvas");
+    var unclaimedctx = unclaimedBoard.getContext("2d");
+    unclaimedctx.clearRect(0, 0, unclaimedBoard.width, unclaimedBoard.height);
+    var claimedBoard1 = document.getElementById("player1canvas");
+    var player1ctx = claimedBoard1.getContext("2d");
+    player1ctx.clearRect(0, 0, claimedBoard1.width, claimedBoard1.height);
+    var claimedBoard2 = document.getElementById("player2canvas");
+    var player2ctx = claimedBoard2.getContext("2d");
+    player2ctx.clearRect(0, 0, claimedBoard2.width, claimedBoard2.height);
+    var stringArray1 = document.getElementById('player1claimed').innerHTML.split('');
+    var stringArray2 = document.getElementById('player2claimed').innerHTML.split('');
+    var player1count = 1;
+    var player2count = 1;
+    var unclaimedCount = 1;
+    for(var i = 0; i < colorArray.length; i++) {
+      if (stringArray1.indexOf(colorArray[i]) != -1) {
+        drawClaimedPiece(colorArray[i], 1, player1count);
+        player1count += 1;
+      }
+      else if (stringArray2.indexOf(colorArray[i]) != -1) {
+        drawClaimedPiece(colorArray[i], 2, player2count);
+        player2count += 1;
+      }
+      else {
+        drawUnclaimedPiece(colorArray[i], unclaimedCount);
+        unclaimedCount += 1;
+      }
+    }
+  }
+
+  function stackGamePiece(xcenter, ycenter, color, stackCount, ctx) {
+    stackOffset = (stackCount - 1) * 5;
+    drawGamePiece(xcenter, ycenter - stackOffset, color, ctx);
+  }
+
+  function shuffleArray(input) {
+    var inputArray = input;
+    for(var i = inputArray.length; i > 0; i--) {
+      randomIndex = Math.random() * (i - 1);
+      x = inputArray[randomIndex];
+      inputArray[randomIndex] = inputArray[i];
+      inputArray[i] = x;
+    }
+    return inputArray;
+  }
+
+  function translateBoardString(boardString, ctx) {
+    var boardArray = boardString.split('|');
+    for (var i = 0; i < boardArray.length; i++) {
+      for (var j = 0; j < boardArray[i].length; j++) {
+        var boardIndex = nodeList[i];
+        var color = letterToColor[boardArray[i][j]];
+        if (color != "Empty") {

```

Comment:
Small point, but if the color == empty you can break out of the innermost loop...